### PR TITLE
Websocket functionalities

### PR DESCRIPTION
**Pull request by rete**
BEGINRELEASENOTES
- Added asio and websocketpp header sources as part of the package
- Implemented WebSocketServer
    - Mimic the interface of the DIM server with Service, Command and Rpc patterns
    - Removed option in CMakeLists.txt to make the WS component optional. Now always built
    - Added test ws server to test the front end connexion
- Removed outdated html/css/js sources

ENDRELEASENOTES

### CODE REVIEW

File: source/include/dqm4hep/WebSocketServer.h [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      void onClose(connection_hdl hdl);
+      void onMessage(connection_hdl hdl, message_ptr msg);
+      void send(WsService *service, const char *buffer, size_t size, bool containsBinary);
+      void onServiceMessage(const std::string &serviceName, connection_hdl hdl, message_ptr msg);
+      void onCommandMessage(const std::string &serviceName, connection_hdl hdl, message_ptr msg);
+      void onRpcMessage(const std::string &serviceName, connection_hdl hdl, message_ptr msg);
+
+    private:
+      connection_map             m_serviceConnections = {};
+      server                     m_server;
```
Comment (apingault) [inline source/include/dqm4hep/WebSocketServer.h]: No default initializer for this type?

File: source/src/WebSocketServer.cc [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    WsService *WsServer::findService(const std::string &name) {
+      auto findIter = m_serviceMap.find(name);
+      if(findIter == m_serviceMap.end()) {
+        return nullptr;
+      }
+      if(findIter->second->type() != SERVICE_TYPE) {
```
Comment (apingault) [inline source/src/WebSocketServer.cc:189]: You could still send a message saying you found something with the right name but the wrong Type.
It means there must be a problem, either you should not be looking for a service or you are looking for the wrong name.

File: source/src/WebSocketServer.cc [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+
+      // Initialize Asio
+      m_server.init_asio();
+
+      // Register our message handler
+      m_server.set_message_handler(std::bind(&WsServer::onMessage, this, _1, _2));
+      m_server.set_open_handler(std::bind(&WsServer::onOpen, this, _1));
+      m_server.set_close_handler(std::bind(&WsServer::onClose, this, _1));
+
+      // Listen on port 9002
```
Comment (apingault) [inline source/src/WebSocketServer.cc]: why 9002? ;)

Comment (rete) [reply source/src/WebSocketServer.cc]: Why not dude !

Comment (apingault) [reply source/src/WebSocketServer.cc]: Just that the comment says 9002 but you are actually listening on `m_port`

**Comment by rete** [comment]
I'm wondering, maybe I should move the new `net::WsServer` to the normal `net::Server` class and run a websocket server in the `net::Server` directly instead of duplicating the classes there.
@apingault What do you think ?

**Comment by rete** [comment]
@apingault I'll merge this when TravisCI is happy

**Pull request merged**